File: Parallel and Concurrent Programming 3/Lab8/Initial codes/Code/Code/ClientServer2/Python/EchoServerTCP.py
import socket

class ServerProtocol:

    def processRequest(self,theInput):
        print("Received message from client: " + theInput)
        theOutput = theInput
        print("Send message to client: " + theOutput)
        return theOutput


class EchoServerTCP:
    #Server address
    PORT = 1234
    serverAdd = ("localhost",PORT)
    EXIT = "CLOSE"

    def main():
        connectionSocket = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
        connectionSocket.bind(EchoServerTCP.serverAdd)
        connectionSocket.listen(10)
        print("Server is listening to port: " + str(EchoServerTCP.PORT))

        conn, add = connectionSocket.accept()
        print("Received request from " + str(add))

        inmsg = conn.recv(1024)

        app = ServerProtocol()
        outmsg = app.processRequest(inmsg.decode())
        while(not(outmsg == EchoServerTCP.EXIT)):
            conn.sendall(outmsg.encode())
            inmsg = conn.recv(1024)
            outmsg = app.processRequest(inmsg.decode())

        conn.close()
        print("Data socket closed")

if __name__ == '__main__':
    EchoServerTCP.main()

File: Parallel and Concurrent Programming 3/Lab8/Task codes/Task1/EchoServerTCP.java
import java.net.*;
import java.io.*;

public class EchoServerTCP {
	private static final int PORT = 1234;
	private static final String EXIT = "CLOSE";

	public static void main(String args[]) throws IOException {

		ServerSocket connectionSocket = new ServerSocket(PORT);
		System.out.println("Server is listening to port: " + PORT);

		Socket dataSocket = connectionSocket.accept();
		System.out.println("Received request from " + dataSocket.getInetAddress());

		//Set input and output streams
		InputStream is = dataSocket.getInputStream();
		BufferedReader in = new BufferedReader(new InputStreamReader(is));
		OutputStream os = dataSocket.getOutputStream();
		PrintWriter out = new PrintWriter(os,true);
		
		//Receive request message, procces it and send reply back
		String inmsg, outmsg;

        ServerProtocol app = new ServerProtocol();

        while (true) {
            inmsg = in.readLine();
            
            if (inmsg.equals("CLOSE")) { // Check for "CLOSE" or null
                break;
            }
            
            outmsg = app.processRequest(inmsg);
            out.println(outmsg);
        }

        dataSocket.close();
        System.out.println("Data socket closed");

		
	}
}			

//Console Results:

// java EchoServerTCP.java
// Server is listening to port: 1234
// Received request from /127.0.0.1
// Received message from client: Nikos,1
// Send message to client: NIKOS
// Received message from client: Nikos,2
// Send message to client: nikos
// Received message from client: Nikos Sak,3,5
// Send message to client: Snptx Xfp
// Received message from client: Snptx Xfp,4,5
// Send message to client: Nikos Sak